pandas/tests/test_join.py

```diff
@@ -348,3 +348,19 @@ def test_merge_join_categorical_multiindex():
     result = a.join(b, on=["Cat1", "Int1"])
     expected = expected.drop(["Cat", "Int"], axis=1)
     assert_frame_equal(expected, result)
+
+
+def test_left_index_and_right_index_true():
+    # From issue 28189
+
+    pdf = DataFrame({"idx": Categorical(["1"] * 4), "value": [1, 2, 3, 4]})
```

Comment:
Construction here is easier if you don't use a dict. should just be able to do:

```python
pd.DataFrame(range(4), columns=["value"], index=pd.Index(pd.Categorical(["1"] * 4), name="idx"))
```

(I changed 1, 2, 3, 4 to 0, 1, 2, 3 in this example but otherwise equivalent)

Reply:
Done!